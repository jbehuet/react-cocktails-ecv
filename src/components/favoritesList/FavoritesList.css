.favorites {
  list-style: none;
  margin: 0;
  padding: 0;
  color: #1f3254;
}

.favorites-item {
  line-height: 35px;
  border-bottom: 1px solid #ccc;
  padding: 0 20px;
}

.favorites-item > span {
  float: right;
  cursor: pointer;
}

.favorites-item > a {
  color: #00a896;
  text-decoration: none;
}
