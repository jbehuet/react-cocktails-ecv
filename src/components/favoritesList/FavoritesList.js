import { Link } from 'react-router-dom';
import './FavoritesList.css';

export function FavoritesList({ favorites, onRemove }) {

    return (
        <ul className="favorites">
            {favorites.map(favorite =>
                <li key={favorite.id} className="favorites-item">
                    <Link to={`/cocktail/${favorite.id}`}>{favorite.name}</Link>
                    <span onClick={() => onRemove(favorite.id)}><i className="fas fa-trash"></i></span>
                </li>
            )}
        </ul>
    )
}