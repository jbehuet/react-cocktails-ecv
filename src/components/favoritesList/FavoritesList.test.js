import React from "react";
import { create } from "react-test-renderer";
import { MemoryRouter } from 'react-router'

import { FavoritesList } from "./FavoritesList";

const fakeFavorites = [{
  "id": "13190",
  "name": "Kool-Aid Shot"
}];

describe("Favorites List component", () => {
  it("renders correctly without favorites", () => {
    const component = create(
      <MemoryRouter>
        <FavoritesList favorites={[]} onRemove={() => { }} />
      </MemoryRouter>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("renders correctly with favorites", () => {
    const component = create(
      <MemoryRouter>
        <FavoritesList favorites={fakeFavorites} onRemove={() => { }} />
      </MemoryRouter>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
