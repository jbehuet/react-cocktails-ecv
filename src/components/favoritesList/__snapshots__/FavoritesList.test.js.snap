// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites List component renders correctly with favorites 1`] = `
<ul
  className="favorites"
>
  <li
    className="favorites-item"
  >
    <a
      href="/cocktail/13190"
      onClick={[Function]}
    >
      Kool-Aid Shot
    </a>
    <span
      onClick={[Function]}
    >
      <i
        className="fas fa-trash"
      />
    </span>
  </li>
</ul>
`;

exports[`Favorites List component renders correctly without favorites 1`] = `
<ul
  className="favorites"
/>
`;
